dotcom-rendering/src/lib/alreadyVisited.ts

```diff
@@ -1,18 +1,28 @@
+import { onConsent } from '@guardian/consent-management-platform';
+
+/**
+ * This local storage item is used to target ads if a user has the correct consents
+ */
 const AlreadyVisitedKey = 'gu.alreadyVisited';
 
-export const getAlreadyVisitedCount = (): number => {
+const getAlreadyVisitedCount = (): number => {
 	const alreadyVisited = parseInt(
 		localStorage.getItem(AlreadyVisitedKey) ?? '',
 		10,
 	);
 	return !Number.isNaN(alreadyVisited) ? alreadyVisited : 0;
 };
 
-export const setAlreadyVisited = (count: number): void => {
+const setAlreadyVisited = (count: number): void => {
 	localStorage.setItem(AlreadyVisitedKey, count.toString());
 };
 
-export const incrementAlreadyVisited = (): void => {
-	const alreadyVisited = getAlreadyVisitedCount();
-	setAlreadyVisited(alreadyVisited + 1);
-};
+export const incrementAlreadyVisited = (): Promise<void> =>
+	onConsent().then((consentState) => {
+		if (consentState.canTarget) {
+			const alreadyVisited = getAlreadyVisitedCount();
+			setAlreadyVisited(alreadyVisited + 1);
+		} else {
+			localStorage.removeItem(AlreadyVisitedKey);
+		}
+	});
```

Comment:
I think `async` functions are easier to read and can remove one level of nesting:

```suggestion
export const incrementAlreadyVisited = async (): Promise<void> =>
    const { canTarget } = await onConsent();
	if(canTarget) {
		const alreadyVisited = getAlreadyVisitedCount();
		setAlreadyVisited(alreadyVisited + 1);
	} else {
		localStorage.removeItem(AlreadyVisitedKey)
    }
```

Reply:
👍 done